sys/riotboot/flashwrite.c

```diff
@@ -150,6 +151,34 @@ int riotboot_flashwrite_putbytes(riotboot_flashwrite_t *state,
     return 0;
 }
 
+int riotboot_flashwrite_invalidate(int slot)
+{
+    extern const unsigned riotboot_slot_numof;
+
+    if (riotboot_slot_numof == 1){
+        LOG_WARNING(LOG_PREFIX "Only one slot configured\n");
+        return -1;
+    }
+    if ((riotboot_slot_validate(riotboot_slot_other()) != 0) || (riotboot_slot_validate(riotboot_slot_current()) != 0)){
+        LOG_INFO(LOG_PREFIX "There will be no valid images to run after reboot\n");
+        return -2;
+    }
+
+    uint8_t data_flash[4] = { INVALIDATE_HDR };
+
+    flashpage_write_raw((void *) riotboot_slot_get_hdr(slot), (const void *) data_flash, 4);
+
+    return 0;
+}
+
+int riotboot_flashwrite_update_revert(void)
+{
+    int _slot_to_revert = -1;
+    _slot_to_revert = (riotboot_slot_get_hdr(riotboot_slot_other())->version
+            > riotboot_slot_get_hdr(riotboot_slot_current())->version) ? riotboot_slot_other() : riotboot_slot_current();
```

Comment:
Is checking the version enough? Should it check if the version is valid?

Reply:
No checking the version number is not enough, that's right. The version validation is done after selecting the slot to revert, using the **riotboot_flashwrite_invalidate** API